modules/log/log.go

```diff
@@ -174,7 +177,7 @@ type logMsg struct {
 // it can contain several providers and log message into all providers.
 type Logger struct {
 	adapter string
-	lock    sync.Mutex
+	lock    sync.RWMutex
```

Comment:
Fair enough. Typically the idiomatic way would be to wrap the map in a new type with its own lock. Now that we have syncmap, we can just use it.